services/obs/obs.service.js

```diff
@@ -0,0 +1,94 @@
+import Joi from 'joi'
+import { BaseXmlService } from '../index.js'
+import { optionalUrl } from '../validators.js'
+import { isBuildStatus, renderBuildStatusBadge } from './obs-build-status.js'
+
+const schema = Joi.object({
+  status: Joi.object({
+    '@_code': isBuildStatus,
+  }).required(),
+}).required()
+
+export default class ObsService extends BaseXmlService {
+  static category = 'build'
+  static route = {
+    base: 'obs',
+    pattern: ':project/:packageName/:repository/:arch',
+    queryParamSchema: Joi.object({
+      instance: optionalUrl,
+    }).required(),
+  }
+
+  static auth = {
+    userKey: 'obs_userName',
+    passKey: 'obs_userPass',
+    authorizedOrigins: [],
+    isRequired: true,
+  }
+
+  static examples = [
+    {
+      title: 'openSUSE OBS package build status',
+      namedParams: {
+        project: 'openSUSE:Tools',
+        packageName: 'osc',
+        repository: 'Debian_11',
+        arch: 'x86_64',
+      },
+      staticPreview: this.render({
+        repository: 'Debian_11',
+        status: 'succeeded',
+      }),
+      keywords: ['open build service'],
+    },
+  ]
+
+  static defaultBadgeData = { label: 'OBS' }
+
+  static render({ repository, status }) {
+    return renderBuildStatusBadge({ repository, status })
+  }
+
+  constructor(context, config) {
+    super(context, config)
+    const origins = []
+    config.public.services.obs.authorizedOrigins.forEach(origin => {
+      const { protocol, host } = new URL(origin)
+      origins.push(`${protocol}//${host}`)
+    })
+    this.authHelper._authorizedOrigins = origins
+  }
+
+  async fetch({ instance, project, packageName, repository, arch }) {
+    return this._requestXml(
+      this.authHelper.withBasicAuth({
+        schema,
+        url: `${instance}/build/${project}/${repository}/${arch}/${packageName}/_status`,
+        errorMessages: {
+          401: 'Not authorized',
+          404: 'Package not found', // extend this to project/package/repository/arch?
```

Comment:
it does return error information. can I use that somehow?

```xml
<status code="unknown_project">
  <summary>Project not found: openSUSE:dTools</summary>
</status>

---

<status code="unknown_package">
  <summary>Package not found: openSUSE:Tools/osd</summary>
</status>

---

<status code="404">
  <summary>project 'openSUSE:Tools' has no repository 'dDebian_11'</summary>
  <details>404 project 'openSUSE:Tools' has no repository 'dDebian_11'</details>
</status>

---

<status code="404">
  <summary>repository 'openSUSE:Tools/Debian_11' has no architecture 'x8d6_64'</summary>
  <details>404 repository 'openSUSE:Tools/Debian_11' has no architecture 'x8d6_64'</details>
</status>
```

or just remove it?

Reply:
Great question. At the moment I don't think we can leverage it since our internal request handling logic will bail when the response status code is a 404, and while it does allow service classes to control the message in the 404 case, it doesn't currently support any additional logic/inspection of the response. That could be an interesting future enhancement for us to consider, but for the purposes of this PR let's just use what's available today and either have the generic/encompassing error or remove the mapping and let the default "not found" apply.

If we do keep a custom message here though, let's be sure to make it all lower case

Reply:
I think "not found" is fitting, so I've just the http status mapping block.